editor/post-permalink/index.js

```diff
@@ -42,21 +46,68 @@ class PostPermalink extends Component {
 		}, 4000 );
 	}
 
+	onEditPermalink( event ) {
+		event.preventDefault();
+		this.setState( { editingSlug: true } );
+	}
+
+	onCancelEditPermalink( event ) {
+		event.preventDefault();
+		this.setState( { editingSlug: false } );
+	}
+
+	onSavePermalink( event ) {
+		event.preventDefault();
+		this.setState( { editingSlug: false } );
+	}
+
 	render() {
-		const { isNew, link } = this.props;
+		const { isNew, link, samplePermalink } = this.props;
+		const { showCopyConfirmation, editingSlug } = this.state;
 		if ( isNew || ! link ) {
 			return null;
 		}
 
+		let permalink = link,
+			viewLink = link;
+		if ( samplePermalink ) {
+			permalink = samplePermalink[ 0 ].replace( '%postname%', samplePermalink[ 1 ] );
+			viewLink += '&preview=true';
```

Comment:
Can we know for certain that the `viewLink` already includes other query parameters? Maybe we ought to use `addQueryArgs` from the `@wordpress/url` package?

https://github.com/WordPress/packages/blob/master/packages/url/README.md